### PR TITLE
🔒 Match ABI Decoder V2's behavior in SafeTransferLib

### PR DESCRIPTION
**Pull request by transmissions11**
fixes https://github.com/Rari-Capital/solmate/issues/141

### CODE REVIEW

File: src/utils/SafeTransferLib.sol [hunk 1]
```diff
@@ -34,105 +33,104 @@ library SafeTransferLib {
...
 
-            // Write the abi-encoded calldata to memory piece by piece:
-            mstore(freeMemoryPointer, 0x23b872dd00000000000000000000000000000000000000000000000000000000) // Begin with the function selector.
-            mstore(add(freeMemoryPointer, 4), and(from, 0xffffffffffffffffffffffffffffffffffffffff)) // Mask and append the "from" argument.
-            mstore(add(freeMemoryPointer, 36), and(to, 0xffffffffffffffffffffffffffffffffffffffff)) // Mask and append the "to" argument.
-            mstore(add(freeMemoryPointer, 68), amount) // Finally append the "amount" argument. No mask as it's a full 32 byte value.
+            // Write the abi-encoded calldata into memory, beginning with the function selector.
+            mstore(freeMemoryPointer, 0x23b872dd00000000000000000000000000000000000000000000000000000000)
+            mstore(add(freeMemoryPointer, 4), from) // Mask and append the "from" argument.
+            mstore(add(freeMemoryPointer, 36), to) // Mask and append the "to" argument.
```
Comment (Joeysantoro) [inline src/utils/SafeTransferLib.sol:45]: comments say mask and append but didn't you remove the mask?

Comment (transmissions11) [reply src/utils/SafeTransferLib.sol:45]: oop

**Review by z0r0z — COMMENTED** [review]
for safeTransferETH, `callStatus` should be `success` to conform to others?

File: src/test/utils/weird-tokens/ReturnsGarbageToken.sol [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    event Transfer(address indexed from, address indexed to, uint256 amount);
+
+    event Approval(address indexed owner, address indexed spender, uint256 amount);
+
+    /*///////////////////////////////////////////////////////////////
+                             METADATA STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    string public constant name = "ReturnsTooLittleToken";
```
[the remark this answers is not in the document]
Comment (transmissions11) [reply src/test/utils/weird-tokens/ReturnsGarbageToken.sol:17]: Oops ty

**Pull request merged**